### PR TITLE
saml2aws: 2.28.0 -> 2.30.0

### PR DESCRIPTION
**Pull request by ari-becker**
###### Motivation for this change
Upstream moved to goreleaser, with some flag changes: https://github.com/Versent/saml2aws/blob/beaaece5fe57049eaa20ed227e9a966e24f268c1/.goreleaser.yml

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ari-becker** [comment]
Closes https://github.com/NixOS/nixpkgs/pull/115558

File: pkgs/tools/security/saml2aws/default.nix [hunk 1]
```diff
@@ -2,24 +2,26 @@
...
     owner = "Versent";
     repo = "saml2aws";
     rev = "v${version}";
-    sha256 = "sha256-2t1MytLjAxhVVsWyMYcQZ9c+ox+X2OszG5mLAv8c7xE=";
+    sha256 = "0plwh7dswf756xy048dqycc2nlpnzk3d1s9m6ypgi77vydryzhj3";
   };
 
   runVend = true;
-  vendorSha256 = "sha256-8Kox01iyWhv/Fp7jHPeNXxc/K2TT1WPyWFieHZkqLho=";
+  vendorSha256 = lib.fakeSha256;
```
Comment (SuperSandro2000) [inline pkgs/tools/security/saml2aws/default.nix:15]: Just try to build it once and it tells you the hash.

**Comment by ari-becker** [comment]
Still has an issue with `buildFlagsArray` not being correctly set, I guess?
This is besides trying to get it to build on OS X, which is why the original 2.28.0 -> 2.28.1 failed

**Comment by ari-becker** [comment]
Obsolete

**Pull request closed without merging**